client/src/auth/sign-in.tsx

```diff
@@ -58,7 +58,7 @@ export default function SignInApp() {
       {userData ? (
         userData.isAuthenticated ? (
           <form
-            className="sign-out-form"
+            className={styles.signOutForm}
```

Comment:
This right here is what's it's all about. I love that. 

What I don't love is that TS doesn't seem to work. I don't get autocomplete on it. 
<img width="621" alt="Screen Shot 2021-06-04 at 1 37 13 PM" src="https://user-images.githubusercontent.com/26739/120842068-3bfacf80-c53a-11eb-8ebc-e82250770099.png">

I know that's supposed to work because I use that in [my `preact-cli` side project](https://github.com/peterbe/groce/). To be honest, I just got whatever `preact-cli` gave me and I've never had a chance to understood how it really works. But if type `import * as style from "./style.css";` and then tab complete from `style.[TAB]`.

<img width="700" alt="Screen Shot 2021-06-04 at 1 41 14 PM" src="https://user-images.githubusercontent.com/26739/120842421-a875ce80-c53a-11eb-81ad-4d5a149d0c5a.png">
<img width="674" alt="Screen Shot 2021-06-04 at 1 41 45 PM" src="https://user-images.githubusercontent.com/26739/120842429-aa3f9200-c53a-11eb-9a24-073d4ba2e9be.png">

I willing to plead ignorance here. I don't know enough about CSS modules to understand it. I can learn. 
But I would say both need to solve this and we need to get @schalkneethling to approve of the idea too. Especially, from a SCSS authoring point of view. (Thank you for your great PR description with the ups and downs).

The question is; where do you we go from here? Shall we dig a little deeper into your implementation and figure out the tooling to get those `style.css.d.ts` generated so we get full autocomplete. Or is that too much perfection this early in the game.